src/oemof/solph/_models.py

```diff
@@ -352,14 +392,34 @@ def __init__(self, energysystem, discount_rate=None, **kwargs):
                 f"you have to specify the `discount_rate` attribute."
             )
             warnings.warn(msg, debugging.SuspiciousUsageWarning)
+
+        else:
+            pass
         super().__init__(energysystem, **kwargs)
 
     def _add_parent_block_sets(self):
         """Add all basic sets to the model, i.e. NODES, TIMESTEPS and FLOWS.
         Also create sets PERIODS and TIMEINDEX used for multi-period models.
         """
-        # set with all nodes
-        self.NODES = po.Set(initialize=[n for n in self.es.nodes])
+        if self.iscellular:
+            # collect all nodes from the child cells
+            self.nodes = self.es.nodes
```

Comment:
I guess, you could move this line above the if condition (line 403) and then you would not need an else statement, would you?

Reply:
Yes, you are correct. Changed that!